public/app/features/transformers/timeSeriesTable/timeSeriesTableTransformer.ts

```diff
@@ -125,3 +136,35 @@ function getLabelFields(frames: DataFrame[]): Record<string, Record<string, Fiel
 
   return labelFields;
 }
+
+function calculateFrameValue(frame: DataFrame, valueType: ValueType): number | null {
+  const valueField = frame.fields.find((field) => field.type === FieldType.number);
+  if (!valueField) {
+    return null;
+  }
+  switch (valueType) {
+    case ValueType.Last:
+      return valueField.values[valueField.values.length - 1] ?? null;
+    case ValueType.Average:
+      const [sum, count] = valueField.values.reduce(
+        ([sum, count], value) => {
+          if (!Number.isNaN(value)) {
+            return [sum + value, count + 1];
+          }
+          return [sum, count];
+        },
+        [0, 0]
+      );
+      return sum / count;
+    case ValueType.Median:
+      const sortedValues = valueField.values.filter((value) => !Number.isNaN(value)).sort((a, b) => a - b);
+      if (sortedValues.length > 0) {
+        if (sortedValues.length % 2 === 0) {
+          return (sortedValues[sortedValues.length / 2 - 1] + sortedValues[sortedValues.length / 2]) / 2;
+        }
+        const middleIndex = Math.floor(sortedValues.length / 2);
+        return sortedValues[middleIndex];
+      }
+      return null;
+  }
+}
```

Comment:
I think you should just be able reuse our existing reducers here, fieldReducers.get(valueType)  , make valueType a  ReducerID

Reply:
ahh, missed this thingys existence :D done!